pageserver/src/tenant/mgr.rs

```diff
@@ -60,6 +61,21 @@ impl TenantsMap {
     }
 }
 
+/// This is "safe" in that that it won't leave behind a partially deleted directory
+/// at the original path, because we rename with TEMP_FILE_SUFFIX before starting deleting
+/// the contents.
+///
+/// This is pageserver-specific, as it relies on future processes after a crash to check
+/// for TEMP_FILE_SUFFIX when loading things.
+async fn safe_remove_tenant_dir_all(path: impl AsRef<Path>) -> std::io::Result<()> {
+    let tmp_path = path_with_suffix_extension(&path, TEMP_FILE_SUFFIX);
+    fs::rename(&path, &tmp_path).await?;
+    if let Some(parent) = path.as_ref().parent() {
+        fs::File::open(parent).await?.sync_all().await?;
+    }
```

Comment:
Would rather we don't even start the renaming if there is no parent. There's always a parent in the places where we call this function.

Reply:
48c073887a16aba0caaae8658939192558c28b2f